zephyr/src/supp_mgmt.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef __SUPP_MGMT_H__
+#define __SUPP_MGMT_H__
+
+/* Management part definitions */
+#include <net/net_mgmt.h>
+#include <net/wifi.h>
+
+#define _NET_SUPP_LAYER NET_MGMT_LAYER_L2
```

Comment:
After some consideration with @wbober we're starting to wonder if there's even a point introducing a supplicant-specific net mgmt layer in NCS.

The core purpose of wifi_mgmt layer was to abstract the MGMT API , so that it could be implemented for various devices. That way an application written with WiFi in mind could use this abstraction to implement an app portable to different devices.

Here however, we're introducing another abstraction layer, that will work only with supplicant and nRF devices. That doesn't make much sense on the second thought.

Perhaps, instead of increasing the complexity with yet another abstraction layer, the application could call the `wpa_supplicant` directly? That way, we would not put on ourselves any constraints applied by the wifi_mgmt API, and we'd avoid extra complexity of introducing yet another network management layer.

In the future, we might even consider implementing a minimalistic wifi_mgmt layer, based on the upstream APIs, on top of the `wpa_supplicant` API. Yes, the API might have less capabilities then we expect from it (unless we want to extend it), but at least we'd retain compability with the upstream at low cost.